Library/Homebrew/dependency_collector.rb

```diff
@@ -5,6 +5,12 @@
 require "requirements"
 require "set"
 
+if OS.mac?
+  require "extend/os/mac/dependency_collector"
+elsif OS.linux?
+  require "extend/os/linux/dependency_collector"
+end
```

Comment:
This should live in `extend/dependency_collector.rb` like the other similar files.